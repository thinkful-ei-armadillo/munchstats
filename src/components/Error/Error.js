import React, { Component } from 'react';
import UserContext from '../../contexts/UserContext';
import './Error.css';

export default class Error extends Component {
  static contextType = UserContext;

  render(){
    return(
      <>
        {this.context.error && (
          <div className='center errorText'>
            <p>{this.context.error}</p>
          </div>
        )}
      </>
    );
  }
}