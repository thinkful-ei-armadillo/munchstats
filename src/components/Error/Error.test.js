import React from 'react';
import Error from './Error';
import {MemoryRouter} from 'react-router-dom';
import {mount} from 'enzyme';
import renderer from 'react-test-renderer';

describe('<Error />', () => {
  it('Renders without crashing', () => {
    mount(<MemoryRouter><Error /></MemoryRouter>);
  });
  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<MemoryRouter><Error/></MemoryRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
});