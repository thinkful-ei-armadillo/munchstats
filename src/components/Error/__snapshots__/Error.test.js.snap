// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Error /> renders the UI as expected 1`] = `null`;
